File: twoSum/two_sum.py
def two_sum(numbers: list[int], target: int):
    hashmap = dict()

    for i, number in enumerate(len(numbers)):
        diff = target - number
        map_value = hashmap.get(diff, None)

        if map_value is not None:
            return [i, map_value]

        hashmap.update({diff: i})

    return []

File: twoSum/twoSum.ts
// Given an array of integers nums and an integer target, return indices of the two numbers such that they add up to target.

// You may assume that each input would have exactly one solution, and you may not use the same element twice.

// You can return the answer in any order.

function twoSum(nums: number[], target: number): number[] {
  const map = new Map();

  for (let i = 0; i < nums.length - 1; i++) {
    const number = nums[i];
    const diff = target - number;
    const mapValue = map.get(diff);

    if (null != mapValue) return [i, mapValue];

    map.set(number, i);
  }

  return [];
}
